datamodel/odata/odata-core/src/main/java/com/sap/cloud/sdk/datamodel/odata/helper/ODataEntitySerializer.java

```diff
@@ -122,6 +123,143 @@ static String serializeEntityForUpdatePatch(
+     * @return The serialized JSON string for entity update request.
+     */
+    @Nonnull
+    static String serializeEntityForUpdatePatchRecursiveDelta(
+        @Nonnull final VdmEntity<?> entity,
+        @Nonnull final Collection<FieldReference> includedFields )
+    {
+        final JsonObject fullEntityJson = GSON_SERIALIZING_NULLS.toJsonTree(entity).getAsJsonObject();
+        final JsonObject patchObject = new JsonObject();
+
+        // Recursively build patch object from changed fields
+        final JsonObject tempPatchObject = createPatchObjectRecursiveDelta(entity, fullEntityJson);
+
+        // Add included fields (from the root only)
+        includedFields
+            .stream()
+            .map(FieldReference::getFieldName)
+            .forEach(key -> patchObject.add(key, fullEntityJson.get(key)));
+
+        // Merge all fields from the tempPatchObject if not already present
+        tempPatchObject
+            .entrySet()
+            .stream()
+            .filter(entry -> !patchObject.has(entry.getKey()))
+            .forEach(entry -> patchObject.add(entry.getKey(), entry.getValue()));
+
+        log.debug("The following delta object is serialized for update : {}.", patchObject);
+
+        return GSON_SERIALIZING_NULLS.toJson(patchObject);
+    }
+
+    /**
+     * Recursively builds a patch object for a VdmObject by including only changed fields. Complex fields are traversed
+     * recursively.
+     *
+     * @param vdmObject
+     *            the VdmObject (entity or complex) to build the patch from
+     * @param jsonObject
+     *            the full JSON representation of this object
+     * @return a JsonObject that contains only changed fields (including nested changes)
+     */
+    @Nonnull
+    private static
+        JsonObject
+        createPatchObjectRecursiveDelta( @Nonnull final VdmObject<?> vdmObject, @Nonnull final JsonObject jsonObject )
+    {
+        final JsonObject patch = new JsonObject();
+
+        // Process all complex fields
+        vdmObject
+            .toMapOfFields()
+            .entrySet()
+            .stream()
+            .filter(entry -> entry.getValue() instanceof VdmComplex<?>)
+            .map(entry -> {
+                final String fieldName = entry.getKey();
+                final VdmComplex<?> complexField = (VdmComplex<?>) entry.getValue();
+                // Recursively build patch for the complex field
+                final JsonObject childJsonObject =
+                    createPatchObjectRecursiveDelta(complexField, jsonObject.getAsJsonObject(fieldName));
+                return Map.entry(fieldName, childJsonObject);
+            })
+            .filter(entry -> !entry.getValue().isEmpty())
+            .forEach(entry -> patch.add(entry.getKey(), entry.getValue()));
+
+        // Add changed primitive fields
+        vdmObject.getChangedFields().keySet().forEach(key -> patch.add(key, jsonObject.get(key)));
```

Comment:
(Minor)

They don't have to be primitive right?

```suggestion
        // Add explicitly changed fields
        vdmObject.getChangedFields().keySet().forEach(key -> patch.add(key, jsonObject.get(key)));
```